src/main/java/net/sf/jabref/gui/actions/CopyBibTeXKeyAndLink.java

```diff
@@ -0,0 +1,61 @@
+package net.sf.jabref.gui.actions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import net.sf.jabref.JabRefGUI;
+import net.sf.jabref.gui.ClipBoardManager;
+import net.sf.jabref.gui.maintable.MainTable;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.OS;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+/**
+ * This class will copy each selected entry's BibTeX key as a hyperlink to it's url to the clipboard.
+ * In case an entry doesn't have a BibTeX key it will not be copied.
+ * In case an entry doesn't have an url this will only copy the BibTeX key.
+ */
+public class CopyBibTeXKeyAndLink implements BaseAction {
```

Comment:
Please add the suffix "Action" to the class name, since this is the convenient way to name the actions in the actions package.